yt/yt/core/logging/log_manager.cpp

```diff
@@ -503,6 +504,11 @@ class TLogManager::TImpl
             Synchronize(TInstant::Now() + Config_->ShutdownGraceTimeout);
         }
 
+        // For now this is the only way to wait for log writers that perform asynchronous flushes.
+        if (Config_->ShutdownBusyTimeout) {
+            Sleep(Config_->ShutdownBusyTimeout);
+        }
```

Comment:
I don't really like this sleep here. However, I understand that it would require a considerable refactoring to be able to wait for the dynamic table log writer closing. For now, let's at least add a TODO for later.